libwallet/src/vault/pjs.rs

```diff
@@ -39,10 +39,15 @@ impl Signer for Pjs {
 
 impl Account for Pjs {
     fn public(&self) -> impl crate::Public {
-        self.inner
+        let mut key = [0u8; 32];
+
+        let pub_key = self.inner
             .get_selected()
             .expect("an account must be defined")
-            .address()
+            .address();
+
+        key.copy_from_slice(pub_key.as_bytes());
```

Comment:
The returned hex string should be decoded, otherwise I don't think you are getting the desired data.